component-studio-integration/src/main/java/org/talend/sdk/component/studio/model/parameter/ElementParameterCreator.java

```diff
@@ -100,12 +101,36 @@ private void addComponentParameters() {
             SettingsCreator advancedCreator = new SettingsCreator(node, ADVANCED, updateComponentsParameter);
             root.accept(advancedCreator, Metadatas.ADVANCED_FORM);
             parameters.addAll(advancedCreator.getSettings());
-            // If No schema add one
-            if (parameters.stream().noneMatch(p -> EParameterFieldType.SCHEMA_TYPE.equals(p.getFieldType()))) {
-                parameters
-                        .add(mainSettingsCreator.createSchemaParameter(EConnectionType.FLOW_MAIN.getName(), "SCHEMA"));
-            }
+            checkSchemaProperties(node, mainSettingsCreator);
+        }
+    }
 
+    /**
+     * Check whether all required schema settings are created and create ones we still need depending on connectors we
+     * have.
+     * 
+     * @param node
+     * @param mainSettingsCreator
+     */
+    protected void checkSchemaProperties(final INode node, final SettingsCreator mainSettingsCreator) {
+        // Get all schema parameters created for current component
+        final Set<String> schemasPresent = parameters
+                .stream()
+                .filter(p -> EParameterFieldType.SCHEMA_TYPE.equals(p.getFieldType()))
+                .map(p -> p.getContext())
+                .collect(Collectors.toSet());
+        // Get all connectors without schema parameter for them
+        final Set<String> connectorNames = component
+                .createConnectors(node)
+                .stream()
+                .filter(c -> c.getDefaultConnectionType().hasConnectionCategory(IConnectionCategory.FLOW)
+                        && !schemasPresent.contains(c.getName()))
+                .map(c -> c.getName())
+                .collect(Collectors.toSet());
+        // Create schema parameter for each connector without schema parameter
+        for (String connectorWithoutSchema : connectorNames) {
+            parameters.add(mainSettingsCreator.createSchemaParameter(connectorWithoutSchema,
```

Comment:
just a note:
This loop may also be rewritten in following way.
```
connectorNames.forEach(c -> parameters.add(mainSettingsCreator.createSchemaParameter(c, "SCHEMA_" + c)));
```
However, your current code looks clearer for me than a code with "forEach".

Reply:
I would leave that as is, if you don't mind.